adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -366,6 +366,24 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   }
 
   /**
+   * Runs a filter that selects data in the underlying RDD that overlaps several genomic regions.
+   *
+   * @param querys The regions to query for.
+   * @return Returns a new GenomicRDD containing only data that overlaps the
+   *   querys region.
+   */
+  def filterByOverlappingRegions(querys: List[ReferenceRegion]): U = {
+    replaceRdd(rdd.filter(elem => {
+
+	  val regions = getReferenceRegions(elem)
+
+	  querys.map(query => {
+	    regions.exists(_.overlaps(query))
+	  }).reduce((a, b) => a || b)
```

Comment:
I would replace `reduce((a, b) => a || b)` with `fold(false)((a, b) => a || b)` as `reduce` will fail on an empty collection. You can get an empty result from `getReferenceRegions(elem)` with unmapped reads.